composer/utils/misc.py

```diff
@@ -100,3 +112,37 @@ def using_torch_2_0_1() -> bool:
         bool: Return True if current version is greater than or equal to 2.0.1 else False
     """
     return version.parse(torch.__version__) >= version.parse('2.0.1')
+
+
+def validate_save_planner(save_planner: Optional[Any]) -> None:
+    """Checks that ``save_planner`` is an instance of a :class:`~torch.distributed.checkpoint.planner.SavePlanner`.
+
+    Raises:
+        ValueError: If ``save_planner`` is not a :class:`~torch.distributed.checkpoint.planner.SavePlanner`.
+    """
+    from torch.distributed.checkpoint.planner import SavePlanner
+
+    if not isinstance(save_planner, SavePlanner):
+        raise ValueError(
+            (
+                f'save_planner {type(save_planner)} is not a '
+                'torch.distributed.checkpoint.planner.SavePlanner'
+            )
+        )
+
+
+def validate_load_planner(load_planner: Optional[Any]) -> None:
+    """Checks that ``load_planner`` is an instance of a :class:`~torch.distributed.checkpoint.planner.LoadPlanner`.
+
+    Raises:
+        ValueError: If ``load_planner`` is not a :class:`~torch.distributed.checkpoint.planner.LoadPlanner`.
```

Comment:
Can we in-line these functions since they are only used in one place?

Reply:
I think this is clearly a util function and would add a lot of extra lines to the main function that make it harder to read. The file is almost 1k lines and the function is almost 200 lines with a lot `if` statement logic already.